src/emqtt_bench.erl

```diff
@@ -623,8 +623,8 @@ run(Parent, I, N, PubSub, Opts0, AddrList, HostList) ->
                         []
                 end,
 
-    Opts1 = replace_opts(Opts0, [ {ifaddr, shard_addr(N, AddrList)}
-                                , {host, shard_addr(N, HostList)}
+    Opts1 = replace_opts(Opts0, [ {ifaddr, shard_addr(I, AddrList)}
+                                , {host, shard_addr(I, HostList)}
```

Comment:
Ok. I see. 
the N is not updating at all due to newly added `I` in recent commits.